### DISCUSSION TITLE
Help! setup-code.txt is gone.

### DISCUSSION THREAD

**Discussion by ItzXRory** [Pryaxis/TShock#2325] [Q&A] [answered]
On second startup in the console there is "TShock will now disable the initial setup system and remove setup-code.txt as it is no longer needed." please help I don't know how to fix it.

**Comment by bartico6** [comment]
It means you used the /setup command to create an admin account - you now have to use said admin account and the setup system is unnecessary, hence the log message

**Reply by ItzXRory** [reply]
the admin account is gone.. What to do.

**Comment by ItzXRory** [comment]
how to make someone an admin?

**Reply by moisterrific** [reply]
`/user group "name" superadmin`

**Reply by hakusaro** [reply]
Please do not tell people to use this group.

**Reply by moisterrific** [reply]
Yeah, I was going to say owner instead, but saw superadmin in the other comments OP made.

**Reply by hakusaro** [reply]
That’s just because that’s a common term. Superadmin will break features for people.

**Comment by ItzXRory** [comment]
like super admin

**Comment by ItzXRory** [comment]
by console

**Comment by drunderscore** [comment]
The [documentation](https://tshock.readme.io/docs/basic-user-management) here should answer most questions!

**Comment by hakusaro** [comment, marked as answer]
Okay, so here's the process you should follow. If you've lost the setup code and you didn't setup an owner account, you need to do that. You either need to make a new account, or you need to make an existing account an admin. In these examples, replace `[username]` with your account name. You can actually have a different account name and username. The account name is what you registered with. Replace `[password]` with your password.

# Option 1: Change group
If you've already registered on the server, this is a good place to start. Just skip to step 2.

1. Register on the server with `/register`. If you get an error saying that the user already exists, that's okay. Just change the group with step 2.
2. From the server console window, type `/user group [username] owner`.

# Option 2: New account for admin purposes
If you don't have any accounts, you can do this. Run from the server console.
1. Run `/user add [username] [password] owner`.

# Which group?

By default, TShock creates a group called `owner` for you. This group grants you most permissions. Another group, called `superadmin`, exists. This group shouldn't be given out though, particularly if you play on an SSC server. The `superadmin` group by definition has all permissions. This means that you'll bypass the SSC system when using this group. If you need to elevate to superadmin temporarily, you can run `/su` to become superuser. You can also, as the `owner`, run `/sudo [command]` to run a command as superadmin, which will alleviate the problem of being logged in as superadmin for day-to-day use.

**Reply by Zicko90** [reply]
when i want to execute the option 2 i ran in to this
![Uploading Screenshot 2023-07-04 201938.png…]()

**Comment by Zicko90** [comment]
i cant when im trying the option 2  it says user coould not  be added check console for details

**Reply by sgkoishi** [reply]
Try option 1 then.

**Comment by ghost** [comment]
Here is an alternitive the "setup.txt" file is changed to "setup.lock" and im pretty sure that you can change the name to "setup.txt" on any system... then change the ".txt" file to the code you want (the code must have no spaces and be only numbers EX. `14135` please note that **RENAMING IT TO THE CODE WILL NOT WORK!**)
